cmake/Corrosion.cmake

```diff
@@ -1571,6 +1571,10 @@ function(corrosion_add_cxxbridge cxx_target)
             $<BUILD_INTERFACE:${generated_dir}/include>
             $<INSTALL_INTERFACE:include>
     )
+    
+    # cxx generated code is using c++11 features in headers, so propagate c++11 as minimal requirement
+    target_compile_features(${cxx_target} PUBLIC cxx_std_11)
```

Comment:
What is the interaction if we set `PUBLIC cxx_std_11` here, and a user binary sets a different version (lower or higher)? (I'm not a C++ person)

Reply:
It will set minimal requirement for any target that link to `${cxx_target}` to be C++11.

Example:
```cmake
add_executable(app main.cpp)
# At this point app c++ level is default to the generator/compiler (not sure which set the default)
target_link_libraries(app PRIVATE ${cxx_target})
# At this point app c++ level is 11
target_compile_features(app PUBLIC cxx_std_17)
# At this point app c++ level is 17
```